platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -136,7 +137,13 @@ class ONIEUpdater(object):
 
     ONIE_IMAGE_INFO_COMMAND = '/bin/bash {} -q -i'
 
+    PLATFORM_ALWAYS_SUPPORT_UPGRADE = ['x86_64-nvidia_sn2201-r0']
```

Comment:
can you please elaborate (provide some comments) why we need it and why we assumes sn2201 is always upgrade? what does it mean?

Reply:
There is a logic to check whether upgrading FW from ONIE is supported in the current version of ONIE.
This is for legacy systems that can have an old ONIE without upgrading FW supported installed.
On alligator, there is no such case because upgrading FW from ONIE is supported in day one.
I'll update the comments.